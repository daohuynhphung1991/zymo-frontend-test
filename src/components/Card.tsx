import * as React from "react";
import { forwardRef, HTMLAttributes, ReactNode } from "react";
import { Card as CardMui, CardContent } from "@mui/material";
import { makeStyles } from "@mui/styles";

/**
 * This is a Card component.
 */
const Card = forwardRef<HTMLDivElement, CardProps>(
  ({ children, className, ...props }, ref) => {
    const classes = useStyles(props);
    return (
      <CardMui {...props} className={classes.cardWrapper}>
        <CardContent>{children}</CardContent>
      </CardMui>
    );
  }
);

const useStyles = makeStyles({
  cardWrapper: {
    margin: 20,
    width: 480,
  },
});

export interface CardProps extends HTMLAttributes<HTMLDivElement> {
  children?: ReactNode;
}

export default Card;
