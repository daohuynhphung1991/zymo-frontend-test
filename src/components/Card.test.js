import React from "react";
import renderer from "react-test-renderer";
import Card from "./Card";

test("Card Component", () => {
  const component = renderer.create(<Card>This is Card Component</Card>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
