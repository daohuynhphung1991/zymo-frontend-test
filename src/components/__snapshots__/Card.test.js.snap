// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-cardWrapper-1 css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    This is Card Component
  </div>
</div>
`;
